pkg/api/v1/schema.go

```diff
@@ -325,6 +349,14 @@ type VolumeSource struct {
 	// the process of populating that PVC with a disk image.
 	// +optional
 	DataVolume *DataVolumeSource `json:"dataVolume,omitempty"`
+	// ConfigMapSource represents a reference to a ConfigMap in the same namespace.
+	// More info: https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/
+	// +optional
+	ConfigMap *ConfigMapVolumeSource `json:"configMap,omitempty"`
```

Comment:
I did not realize before that the k8s structs for configmaps and secrets use in one case `name` and in the other case `secretName` to reference the k8s object.